### PR TITLE
Dev 533 transform javascript where function to fql

### PR DESCRIPTION
**Pull request by firehudson**

### CODE REVIEW

**Comment by firehudson** [comment]
**Regarding Sonar Issue:** This is expected as we have to use `Function` constructor to convert string function to callable function.

Currently, we are passing a valid fauna query string from client side for `where` predicate function, please check [here](https://github.com/Etesie/fauna-typed/blob/DEV-533-transform-javascript-where-function-to-fql/src/routes/cms/client/table/%2Bpage.svelte#L37-L47). 

To run this predicate function's string in document store to get data from store's state we convert string to function using `Function` constructor, please check [here](https://github.com/Etesie/fauna-typed/blob/DEV-533-transform-javascript-where-function-to-fql/src/lib/stores/document.svelte.ts#L186).

To strictly check of where predictable function's string, we are using ``(item) => ${string}`` typescript type instead of  `string`, please check [here](https://github.com/Etesie/fauna-typed/blob/DEV-533-transform-javascript-where-function-to-fql/src/lib/stores/document.svelte.ts#L34)

If above solution looks good to you @mmailaender , Should we apply `where` method's predicate function type to Collection store's `where` method as well?

File: src/lib/stores/document.svelte.ts [hunk 1]
```diff
@@ -31,7 +31,7 @@ export type CreateDocumentStore<
 	last: () => Functions<T, T_Replace, T_Update>;
 	all: () => Page<Functions<T, T_Replace, T_Update>>;
 	paginate: (after: string) => Page<Functions<T, T_Replace, T_Update>>; // TODO: To be implemented
-	where: (filter: Predicate<Document<T>>) => Page<Functions<T, T_Replace, T_Update>>;
```
Comment (mmailaender) [inline src/lib/stores/document.svelte.ts:34]: Are you aware of what you have done here? Please explain the implications to me.

**Review by mmailaender — CHANGES_REQUESTED** [review]

File: src/lib/stores/document.svelte.ts [hunk 2]
```diff
@@ -175,9 +175,13 @@ export const createDocumentStore = <K extends keyof TypeMapping>(
 					};
 
 				case 'where':
-					return (filter: Predicate<Document<MainType>>) => {
-						const result = new Page(getObjects(filter), undefined);
-						// db.where(filter);
+					return (filter: Predicate<Document<MainType>> | `(item) => ${string}`) => {
```
Comment (mmailaender) [inline src/lib/stores/document.svelte.ts:178]: Bad practice - The API is accepting only `filter: Predicate<Document<MainType>>` - accepting different types inside the proxy will create Type errors.

**Review by mmailaender — CHANGES_REQUESTED** [review]

File: src/lib/stores/document.svelte.ts (hunk 2 above)
Comment (firehudson) [reply src/lib/stores/document.svelte.ts:180]: Okay, let me update `filter: Predicate<Document<MainType>>` to `filter: Predicate<Document<MainType>> | '(item) => ${string}'`. By using this type user will have both options to pass predicate as string or function

Comment (mmailaender) [reply src/lib/stores/document.svelte.ts:180]: There are multiple parties: 
`Client` -> `Fauna Typed API` -> `ETL` -> `Fauna API`, and they should not be mixed.
The API provides a clean and concise typed API. In our case, it expects a function of type `Predicate<Document<MainType>>`, and the ETL takes care of transforming it into an FQL Function. But what you are doing is, allowing the Client to pass a typed Function AND a string. That makes the API overloaded and unclear to a consumer and deprives the ELT of its function.

**Review by mmailaender — CHANGES_REQUESTED** [review]

**Review by mmailaender — APPROVED** [review]

**Pull request merged**